presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayDistinctFromOperator.java

```diff
@@ -47,11 +47,8 @@ public static boolean isDistinctFrom(
             @SqlType("array(E)") Block right,
             @IsNull boolean rightNull)
     {
-        if (leftNull != rightNull) {
-            return true;
-        }
-        if (leftNull) {
-            return false;
+        if (leftNull || rightNull) {
+            return leftNull ^ rightNull;
```

Comment:
It looks like there are twice as many people who voted for `^` compared to those who voted for `!=`. I'm ok.